daemon/dns/dns_test.go

```diff
@@ -0,0 +1,125 @@
+package dns
```

Comment:
it might be good to have a test that checks that all the available commands are all iterated?
and that it stops at the first that doesn't return error?

Reply:
This kind of test i.e. iterate over methods, check availability, handle errors - it is implemented right now. Just it is abstracted from specific dns handling implementations. See `dns_test.go` `Test_Method`